src/AMQP/CQRS/ICommand.ts

```diff
@@ -1,13 +1,13 @@
 
-export interface ICommandPayload {
-	type: string;
+export interface ICommandPayload<T extends string> {
+	type: T;
 }
 
-interface ICommand<TPayload extends ICommandPayload = ICommandPayload> {
+interface ICommand<TPayload extends string> {
```

Comment:
Tohle bych ale právě nedělal ze tří důvodů:

1) TPayload není string, ale nějaký objekt, který musí obsahovat type: string ale může mít více vlastností.
2) Pokud jako typový argument neecháš předat celý payload, tak je to univerzálnější. Takhle se stringem můžeš ovlivnit jen type a ostatní pak nejsou vidět.
3) je to nekompetibilní změna a zrovna tohle už se hodně používá.

Přiklad použítí.
```ts
interface MyCommand extends ICommand<{ type: 'MyCommand', mojeProperty: number }> {}

const command: MyCommand = ...{dej mi myCommand};
command.payload.mojeProperty; // díky generice to tady TS povolí. S tvým řešením by to vyhodilo typovou chybu.
```

Osobně bych to přepsal když už tak takto:
```ts

interface ICommandPayload<TType extends string> {
  type: TType;
}

interface ICommand<TPayload extends ICommandPayload<TType> = ICommandPayload<TType>, TType extends string = string> {
  type: TType;
  id: string | null;
  // atd.
  payload: TPayload;
}
```
Ale nevim jestli to TS už takhle zvládne